### ISSUE TITLE
MapboxDraw.modes is undefined

### ISSUE THREAD

**Issue by cyrilchapon** [mapbox/mapbox-gl-draw#683]
**mapbox-gl-js version**:
**mapbox-gl-draw version**:

### Steps to Trigger Behavior

 1. `console.log(MapboxDraw.modes)`

### Expected Behavior

[API](https://github.com/mapbox/mapbox-gl-draw/blob/master/docs/API.md) states that

> modes, Object: over ride the default modes with your own. MapboxDraw.modes can be used to see the default values

### Actual Behavior

`MapboxDraw.modes` is undefined

**Comment by cyrilchapon** [comment]
Temporary workaround :

Instanciate a temporary object 
```
(new MapboxDraw()).modes
```

**Comment by mcwhittemore** [comment]
`MapboxDraw.modes` should be `./src/modes/index.js` while `api.modes` is the Modes constants object. Part of me wants to drop `api.modes` along with a fix for this to help mitigate confusion about what these two values indicate.

**Comment by cyrilchapon** [comment]
Agreed, furthermore js convention for uppercase is prototype I think, while lowercased camel for instances, isn't it ?

Envoyé de mon iPhone

Le 29 août 2017 à 17:49, Matthew Chase Whittemore <notifications@github.com<mailto:notifications@github.com>> a écrit :


MapboxDraw.modes should be ./src/modes/index.js while api.modes is the Modes constants object. Part of me wants to drop api.modes along with a fix for this to help mitigate confusion about what these two values indicate.

**Comment by mcwhittemore** [comment]
I'm not sure what you're asking @cyrilchapon. Can you provide an example?

**Comment by pedrobarquinha** [comment]
Hey there! @mcwhittemore was this ever fixed? I'm still getting this as undefined and have been unable to add custom modes on the most recent release.

Because of this issue, the instructions provided at https://github.com/mapbox/mapbox-gl-draw/blob/master/docs/MODES.md don't work.
